### PR TITLE
populate redshift courses table from dashboard unit_groups table

### PR DESCRIPTION
**Pull request by davidsbailey**
Follow-on to #35949. To be merged alongside #35952.

Discussion with Infra team: https://codedotorg.slack.com/archives/C03CK49G9/p1596053230287400 

Discussion with data team: https://codedotorg.slack.com/archives/C0SUN2W3D/p1595878192355900

## Testing story

partial output from `bundle exec rake stack:data:validate RAILS_ENV=production VERBOSE=1`:

```
Listing changes to existing stack `DATA-production`:
Modify DMSCronLevelSourcesPii [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (TableMappings)
Modify DMSCronPii [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (TableMappings)
Modify DMSCronUserHierarchyPii [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (TableMappings)
Modify DMSCronUserHierarchy [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (TableMappings)
Modify DMSCronUserLevels [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (TableMappings)
Modify DMSCron [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (TableMappings)
```

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: aws/dms/tasks.yml [hunk 1]
```diff
@@ -47,7 +47,8 @@ cron:
 - dashboard.levels
 - dashboard.script_levels
 - dashboard.levels_script_levels
-- dashboard.courses
+- dashboard.unit_groups
+    rename: courses
```
Comment (davidsbailey) [inline aws/dms/tasks.yml]: Syntax taken from here: https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Tasks.CustomizingTasks.TableMapping.html#CHAP_Tasks.CustomizingTasks.TableMapping.SelectionTransformation.Transformations

**Comment by davidsbailey** [comment]
@wjordan @sureshc can you please take a look?

File: aws/dms/tasks.yml (hunk 1 above)
Comment (sureshc) [reply aws/dms/tasks.yml]: I think the `rename` property here won't trigger creation of a DMS `rename` transformation rule until we update [`Cdo::DMS.tasks`](https://github.com/code-dot-org/code-dot-org/blob/ea7bac4ecd268d886943a59dc6a3da0f90387377/lib/cdo/aws/dms.rb#L13)

I'll push a commit to this branch today that adds that functionality.

Comment (davidsbailey) [reply aws/dms/tasks.yml]: Hi @sureshc , does this seem like something you'll be able to get to this week? If not, I can take a crack at it.

**Comment by sureshc** [comment]
Our DMS Tasks currently prefix each MySQL table with `_import_`. I'm not sure yet whether we can use the DMS Task transformation rules to both rename the source table and prefix it. I'm running an experiment today to check that.

**Comment by davidsbailey** [comment]
any update @sureshc ?

**Comment by sureshc** [comment]
> any update @sureshc ?

I haven't had a chance to test out the renaming functionality, but I think the DMS transformation rules won't support renaming the source table name, and then prefixing the target table name. I'll try to run that test this afternoon. I think we may need to look for alternative solutions, like have the RED (data) team create a view old table name that points to the new table name.

**Comment by davidsbailey** [comment]
ok, great, thanks for the update! creating a view sounds promising if the other approach falls through. I would imagine that the data team would just test it out with a different view name besides "courses", and then once it is working, drop the "courses" table from redshift and rename the view to "courses". they would probably know better than I do about that, though.

**Comment by sureshc** [comment]
@davidsbailey I've committed some logic that I believe will work. I'm manually diffing the JSON Task Mapping rules before/after this proposed change to protect against potential regressions and will post those results today. Can you rebase this feature branch so that it picks up the Drone caching fix?

I checked with RED (data) team. Creating a View called `courses` that points to the new table is an option, but it's an important table used across many complex views, so there would be some risk. Renaming the table through the MySQL --> Redshift process continues to look like the best option.

**Comment by davidsbailey** [comment]
hey @sureshc , rebasing to pick up the drone caching fix seems to have caused drone to pass. what are the next steps here?

**Comment by sureshc** [comment]
> hey @sureshc , rebasing to pick up the drone caching fix seems to have caused drone to pass. what are the next steps here?

Let's sync up after today's production release and before the DMS import at 7PM PDT and run the CloudFormation stack update manually to deploy these new task definitions! Maybe 4pm today?

**Pull request closed without merging**